airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py

```diff
@@ -96,14 +96,45 @@ def _get_refresh_access_token_response(self):
         except Exception as e:
             raise Exception(f"Error while refreshing access token: {e}") from e
 
-    def refresh_access_token(self) -> Tuple[str, int]:
+    def refresh_access_token(self) -> Tuple[str, Union[str, int]]:
         """
-        Returns the refresh token and its lifespan in seconds
+        Returns the refresh token and its lifespan
 
-        :return: a tuple of (access_token, token_lifespan_in_seconds)
+        :return: a tuple of (access_token, token_lifespan)
         """
         response_json = self._get_refresh_access_token_response()
-        return response_json[self.get_access_token_name()], int(response_json[self.get_expires_in_name()])
+
+        return response_json[self.get_access_token_name()], response_json[self.get_expires_in_name()]
+
+    def _parse_token_lifespan(self, value: Union[str, int]) -> pendulum.DateTime:
```

Comment:
nit: how about renaming this `_parse_token_expiration_date` since the returned value is the datetime of its expiration. lifespan makes me think the returned value is the time until it expires